### PR TITLE
Fix electrs and add electrs hidden service

### PR DESCRIPTION
**Pull request by nixbitcoin**
This commit fixes electrs, reenables it, adds an electrs Tor v3 HS so users can connect from the outside, and a high-memory option. 

The high-memory option allows users with high-memory (< 8 GB RAM) to sync faster. How should this be documented?

### CODE REVIEW

File: modules/bitcoind.nix [hunk 1]
```diff
@@ -9,6 +9,9 @@ let
     ${optionalString cfg.testnet "testnet=1"}
     ${optionalString (cfg.dbCache != null) "dbcache=${toString cfg.dbCache}"}
     ${optionalString (cfg.prune != null) "prune=${toString cfg.prune}"}
+    sysperms=${if cfg.sysperms then "1" else "0"}
+    disablewallet=${if cfg.disablewallet then "1" else "0"}
+
```
Comment (jonasnick) [inline modules/bitcoind.nix]: unnecessary newline

File: modules/bitcoind.nix [hunk 2]
```diff
@@ -195,6 +214,7 @@ in {
         chmod o-rw  '${cfg.dataDir}/bitcoin.conf'
         chown '${cfg.user}:${cfg.group}' '${cfg.dataDir}/bitcoin.conf'
         echo "rpcpassword=$(cat /secrets/bitcoin-rpcpassword)" >> '${cfg.dataDir}/bitcoin.conf'
+        chmod -R g+rX '${cfg.dataDir}/blocks'
```
Comment (jonasnick) [inline modules/bitcoind.nix]: is that necessary despite setting sysperms?

File: modules/bitcoind.nix [hunk 3]
```diff
@@ -9,6 +9,9 @@ let
     ${optionalString cfg.testnet "testnet=1"}
     ${optionalString (cfg.dbCache != null) "dbcache=${toString cfg.dbCache}"}
     ${optionalString (cfg.prune != null) "prune=${toString cfg.prune}"}
+    sysperms=${if cfg.sysperms then "1" else "0"}
+    disablewallet=${if cfg.disablewallet then "1" else "0"}
```
Comment (jonasnick) [inline modules/bitcoind.nix]: You can also have both options optional by declaring their type as `types.nullOr types.bool` and then using `${optionalString (cfg.sysperms != null) "sysperms=${if cfg.sysperms then "1" else "0"}"}`.

File: modules/electrs.nix [hunk 4]
```diff
@@ -18,13 +20,20 @@ in {
       default = "/var/lib/electrs";
       description = "The data directory for electrs.";
     };
+    high-memory = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+      If enabled, the electrs service will sync faster on high-memory systems.
```
[the remark this answers is not in the document]
Comment (nixbitcoin) [reply modules/electrs.nix]: ≤ 8GB. Should I put this in the description?

File: modules/bitcoind.nix (hunk 2 above)
Comment (nixbitcoin) [reply modules/bitcoind.nix]: Yes. For blocks synced before sysperm was enabled.

File: modules/bitcoind.nix (hunk 3 above)
Comment (nixbitcoin) [reply modules/bitcoind.nix]: Fixed in fca4af5

File: modules/bitcoind.nix (hunk 1 above)
Comment (nixbitcoin) [reply modules/bitcoind.nix]: Fixed in fca4af5

File: modules/nix-bitcoin.nix [hunk 5]
```diff
@@ -157,14 +158,20 @@ in {
...
     services.spark-wallet.enable = cfg.modules == "all";
-    # TODO: re-enable when fixed
-    services.electrs.enable = false;
     services.tor.hiddenServices.spark-wallet = {
       map = [{
         port = 80; toPort = 9737;
       }];
       version = 3;
     };
+    services.electrs.enable = true;
```
[the remark this answers is not in the document]
Comment (nixbitcoin) [reply modules/nix-bitcoin.nix]: Fixed in fca4af5

File: modules/bitcoind.nix [hunk 6]
```diff
@@ -151,6 +154,22 @@ in {
           If enabled, the bitcoin service will listen.
         '';
       };
+      sysperms = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+        Create new files with system default permissions, instead of umask 077 (only effective with disabled wallet functionality)
+        # Necessary for electrs
```
[the remark this answers is not in the document]
Comment (nixbitcoin) [reply modules/bitcoind.nix]: Fixed in fca4af5

File: modules/electrs.nix (hunk 4 above)
[the remark this answers is not in the document]
Comment (jonasnick) [reply modules/electrs.nix]: yeah

File: modules/bitcoind.nix (hunk 2 above)
Comment (jonasnick) [reply modules/bitcoind.nix]: ok

File: modules/electrs.nix (hunk 4 above)
[the remark this answers is not in the document]
Comment (nixbitcoin) [reply modules/electrs.nix]: Fixed in eb4968d292479edf09ca3e62c2551ef637117900

[the remark this answers is not in the document]
Comment (jonasnick) [reply modules/electrs.nix]: Should be `>= 8GB`? `<= 8GB` means that 1MB is high memory.

**Comment by jonasnick** [comment]
ACK

**Pull request merged**